docs/NativeModulesIOS.md

```diff
@@ -173,6 +173,37 @@ A native module is supposed to invoke its callback only once. It can, however, s
 
 If you want to pass error-like objects to JavaScript, use `RCTMakeError` from [`RCTUtils.h`](https://github.com/facebook/react-native/blob/master/React/Base/RCTUtils.h).  Right now this just passes an Error-shaped dictionary to JavaScript, but we would like to automatically generate real JavaScript `Error` objects in the future.
 
+## Promises
+
+Native modules can also return a Promise, which can simplify your code quite lot, especially when using ES7 `async/await` syntax.
+
+If we refactor the above code using callbacks to use Promise, it becomes the following,
+
+```objective-c
+RCT_EXPORT_METHOD(resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject))
+{
+  NSArray *events = ...
+  if (events) {
+    resolve(events);
+  } else {
+    reject(error);
+  }
+}
+```
+
+This method can then be accessed in JavaScript using:
```

Comment:
Likewise, I have the same suggestion as for Android -- specifically, putting the code inside of an async function so that people don't try to use `await` outside of one, which is nonsensical.